lib/avalon/batch/manifest.rb

```diff
@@ -25,14 +25,51 @@ class Manifest
       SKIP_FIELDS = [:collection]
 
       def_delegators :@entries, :each
-      attr_reader :spreadsheet, :file, :name, :email, :entries, :package
+      attr_reader :spreadsheet, :file, :name, :email, :entries, :package, :manifest_logger
 
       class << self
         def locate(root)
+          locate_with_status(root)[:processable]
+        end
+
+        def locate_with_status(root)
+          status = {
+            processable: [],
+            processing: [],
+            processed: [],
+            error: []
+          }
           possibles = Dir[File.join(root, "**/*.{#{EXTENSIONS.join(',')}}")]
-          possibles.reject do |file|
-            File.basename(file) =~ /^~\$/ or self.error?(file) or self.processing?(file) or self.processed?(file)
+          possibles.each do |file|
+            next if File.basename(file) =~ /^~\$/
+            if self.error?(file)
+              status[:error] << file
+            elsif self.processing?(file)
+              status[:processing] << file
+            elsif self.processed?(file)
+              status[:processed] << file
+            else
+              status[:processable] << file
+            end
           end
+          status
+        end
+
+        def concatenate_log_files(root)
+          manifests = []
+          locate_with_status(root).each { |file_status, files| manifests += files }
```

Comment:
you using locate_with_status but you dont care about file_status at all? So you doing all this calculating of the status arrays but then dont even care.

Just extract the above `possibles = Dir[File.join(root, "**/*.{#{EXTENSIONS.join(',')}}")]`  and any other logic you need (like ignoring file with /^~\$/) `locate_with_status` into a seperate method and call them here and in locate_with_status instead.

at the very least `file_status` should become `_file_status` to indicate your not using it.